tests/test_utils.py

```diff
@@ -188,3 +190,9 @@ def test_consume():
     consume(x)
     with pytest.raises(StopIteration):
         next(x)
+
+
+class TestNamespace:
+    def test_isawaitable(self):
+        """Test async compatibility by asserting the required type check works."""
+        assert not inspect.isawaitable(Namespace())
```

Comment:
Wouldn't it be better to have a test that tests the namespace object in an actual async template context?

Reply:
I wasn't sure how extensive we wanted the test to be, so I decided to keep it small initially. It's now updated with template integration coverage.